src/core/Global.ttslua

```diff
@@ -825,3 +779,146 @@ function urldecode(str)
+    end
+  end
+end
+
+-- handles the applying of option selections and calls the respective functions based
+---@param id String ID of the option that was selected or deselected
+---@param state Boolean State of the option (true = enabled)
+function applyOptionPanelChange(id, state)
+  -- option: Snap tags
+  if id == "useSnapTags" then
+    playmatAPI.setLimitSnapsByType(state, "All")
+
+  -- option: Draw 1 button
+  elseif id == "showDrawButton" then
+    playmatAPI.showDrawButton(state, "All")
+
+  -- option: Clickable clue counters
+  elseif id == "useClueClickers" then
+    playmatAPI.clickableClues(state, "All")
+
+    -- update master clue counter
+    getObjectFromGUID("4a3aa4").setVar("useClickableCounters", state)
+  
+  -- option: Show token arranger
+  elseif id == "showTokenArranger" then
+    -- delete previously pulled out tokens
+    for _, token in ipairs(getObjectsWithTag("to_be_deleted")) do token.destruct() end
+
+    spawnOrRemoveHelper(state, "Token Arranger", {-42.3, 1.4, -46.5})
+
+  -- option: Show clean up helper
+  elseif id == "showCleanUpHelper" then
+    spawnOrRemoveHelper(state, "Clean Up Helper", {-68, 1.6, 35.5})
+  
+  -- option: Show hand helper for each player
+  elseif id == "showHandHelper" then
+    spawnOrRemoveHelper(state, "Hand Helper", {-50.84, 1.6, 7.02}, {0, 270, 0}, "White")
+    spawnOrRemoveHelper(state, "Hand Helper", {-50.90, 1.6, -25.10}, {0, 270, 0}, "Orange")
+    spawnOrRemoveHelper(state, "Hand Helper", {-34.38, 1.6, 22.44}, {0, 000, 0}, "Green")
+    spawnOrRemoveHelper(state, "Hand Helper", {-16.69, 1.6, -22.42}, {0, 180, 0}, "Red")
+
+  -- option: Show chaos bag manager
+  elseif id == "showChaosBagManager" then
+    spawnOrRemoveHelper(state, "Chaos Bag Manager", {-67.8, 1.4, -49.5})
+
+  end
+end
+
+-- handler for spawn / remove functions of helper objects
+---@param state Boolean Contains the state of the option: true = spawn it, false = remove it
+---@param name String Name of the helper object
+---@param position Vector Position of the object (where it will spawn or where it will be removed from)
+---@param rotation Vector Rotation of the object for spawning (default: {0, 270, 0})
+---@param color Color This is only needed for correctly setting the color of the "Hand Helper"
+function spawnOrRemoveHelper(state, name, position, rotation, color)
+  if state then
+    spawnHelperObject(name, position, rotation, color)
+    Player["White"].pingTable(position)
+  else
+    removeHelperObject(name, position)
+  end
+end
+
+-- copies the specified tool (by name) from the barrel
+---@param name String Name of the object that should be copied
+---@param position Position Desired position of the object
+function spawnHelperObject(name, position, rotation, color)
+  if rotation == nil then rotation = {0, 270, 0} end
+
+  local barrel = getObjectFromGUID("aa8b38")
+  local barrelCopy = barrel.clone({position = barrel.getPosition() + Vector(0, 3, 0)})
```

Comment:
Use getData().ContainedObjects and spawnObjectData to avoid having to copy the barrel and do a take

Reply:
That would indeed be a bit cleaner, but I can't get that to work. Data structure that is returned by `getData()` is not clear to me and of course not documentated...

Reply:
Yeah, that is a better solution! Changed it